service/strong_consistency/sc_state_machine.cc

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "sc_state_machine.hh"
+#include "serializer_impl.hh"
+#include "idl/sc_state_machine.dist.hh"
+#include "idl/sc_state_machine.dist.impl.hh"
+#include "replica/database.hh"
+
+namespace service {
+    static logging::logger logger("sc_sm");
+
+    class sc_state_machine: public raft_state_machine {
+        locator::global_tablet_id _tablet;
+        raft::group_id _group_id;
+        replica::database& _db;
+
+    public:
+        sc_state_machine(locator::global_tablet_id tablet,
+            raft::group_id gid,
+            replica::database& db)
+            : _tablet(tablet)
+            , _group_id(gid)
+            , _db(db)
+        {
+        }
+
+        future<> apply(std::vector<raft::command_cref> command) override {
+            try {
+                utils::chunked_vector<frozen_mutation> muts;
+                muts.reserve(command.size());
+                for (const auto& c: command) {
+                    auto is = ser::as_input_stream(c);
+                    auto cmd = ser::deserialize(is, std::type_identity<sc_raft_command>{});
+                    muts.push_back(std::move(cmd.mutation));
+                }
+
```

Comment:
Shouldn't we upgrade mutation in case current schema version is different, or `apply()` does it for us?